### PR TITLE
[llvm] Remove `br i1 undef` from regression tests

### PR DESCRIPTION
**Pull request by leewei05**
This PR removes tests with `br i1 undef` under `llvm/tests/Transforms/ObjCARC, Reassociate, SCCP, SLPVectorizer...`. After this PR, I'll continue to fix tests under `llvm/tests/CodeGen`, which has more UB tests than `llvm/tests/Transforms`.

@nunoplopes @regehr

### CODE REVIEW

File: llvm/test/Transforms/SCCP/2004-12-10-UndefBranchBug.ll [hunk 1]
```diff
@@ -1,14 +1,13 @@
...
-define i32 @foo() {
+define i32 @foo(i1 %arg) {
 ; CHECK-LABEL: @foo(
-; CHECK-NEXT:    unreachable
+; CHECK-NEXT:    br i1 [[ARG:%.*]], label [[T:%.*]], label [[T]]
+; CHECK:       T:
+; CHECK-NEXT:    ret i32 1
 ;
-  br i1 undef, label %T, label %T
+  br i1 %arg, label %T, label %T
```
Comment (nunoplopes) [inline llvm/test/Transforms/SCCP/2004-12-10-UndefBranchBug.ll:10]: leave this one as poison to retain the spirit of the test

File: llvm/test/Transforms/SCCP/return-zapped.ll [hunk 2]
```diff
@@ -1,37 +1,41 @@
 ; NOTE: Assertions have been autogenerated by utils/update_test_checks.py UTC_ARGS: --function-signature
 ; RUN: opt < %s -S -passes=ipsccp | FileCheck %s
 
-; testf() performs an unconditional branch on undef, as such the testf() return
```
Comment (nunoplopes) [inline llvm/test/Transforms/SCCP/return-zapped.ll:4]: pls don't remove this comment nor change the spirit of the test

File: llvm/test/Transforms/SCCP/solve-after-each-resolving-undefs-for-function.ll [hunk 3]
```diff
@@ -26,22 +27,25 @@ if.end:                                          ; preds = %if.then1, %entry
...
+define internal i32 @test1(i1 %c, i1 %arg) {
+; CHECK-LABEL: define internal range(i32 0, 100) i32 @test1(
+; CHECK-SAME: i1 [[C:%.*]], i1 [[ARG:%.*]]) {
 ; CHECK-NEXT:  [[ENTRY:.*:]]
 ; CHECK-NEXT:    br label %[[IF_THEN:.*]]
 ; CHECK:       [[IF_THEN]]:
-; CHECK-NEXT:    [[CALL:%.*]] = call i32 @testf(i1 [[C]])
-; CHECK-NEXT:    br label %[[RET1:.*]]
+; CHECK-NEXT:    [[CALL:%.*]] = call i32 @testf(i1 [[C]], i1 [[ARG]])
+; CHECK-NEXT:    [[RES:%.*]] = icmp eq i32 [[CALL]], 10
```
Comment (nunoplopes) [inline llvm/test/Transforms/SCCP/solve-after-each-resolving-undefs-for-function.ll:37]: don't change the spirit of the test

**Comment by YoungJunLee-NSHC** [comment]
@leewei05
Hello, why did you change it to `br i1 %arg`?

**Comment by IamYJLee** [comment]
@leewei05
Hello, why did you change it to br i1 %arg?

**Pull request merged**